src/v/cluster/health_monitor_frontend.cc

```diff
@@ -147,4 +147,10 @@ ss::future<bool> health_monitor_frontend::does_raft0_have_leader() {
       [](health_monitor_backend& be) { return be.does_raft0_have_leader(); });
 }
 
+ss::future<> health_monitor_frontend::reset() {
+    vlog(clusterlog.info, "Resetting Health Monitor Frontend");
+    co_await stop();
```

Comment:
`stop()` method closed gate. So after `reset()` you will never run again updating health state

Reply:
>stop() method closes the gate. So after reset() you will never run again updating health state.

@NyaliaLui this was the same reentrancy issue i mentioned in that other pr